libcxx/include/__algorithm/shift_left.h

```diff
@@ -24,30 +26,40 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 
 #if _LIBCPP_STD_VER >= 20
 
-template <class _ForwardIterator>
-inline _LIBCPP_HIDE_FROM_ABI constexpr _ForwardIterator
-shift_left(_ForwardIterator __first,
-           _ForwardIterator __last,
-           typename iterator_traits<_ForwardIterator>::difference_type __n) {
+template <class _AlgPolicy, class _Iter, class _Sent>
```

Comment:
Please add a `_LIBCPP_ASSERT_UNCATEGORIZED` to check the precondition (CC @var-const).